cmd/utils/auth.go

```diff
@@ -32,3 +46,69 @@ func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
 
 	return chain, nil
 }
+
+// cachedInteractiveBrowserCredential is a credential that uses the interactive browser to authenticate and caches the token.
+// TODO: This is a workaround until the azidentity package supports caching, https://github.com/Azure/azure-sdk-for-go/issues/16643.
+type cachedInteractiveBrowserCredential struct {
+	client public.Client
+}
+
+func newCachedInteractiveBrowserCredential() (*cachedInteractiveBrowserCredential, error) {
+	file := path.Join(homedir.HomeDir(), ".kubectl-az", "token-cache.json")
+	if err := os.MkdirAll(path.Dir(file), os.ModePerm); err != nil {
+		return nil, fmt.Errorf("creating cache directory: %w", err)
+	}
+
+	client, err := public.New(developerSignOnClientID,
+		public.WithCache(&tokenCache{file: file}),
+		public.WithAuthority(runtime.JoinPaths(string(azidentity.AzurePublicCloud), organizationsTenantID)),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating public client: %w", err)
+	}
+
+	return &cachedInteractiveBrowserCredential{client: client}, nil
+}
+
+// GetToken implements the azcore.TokenCredential interface on cachedInteractiveBrowserCredential.
+func (c *cachedInteractiveBrowserCredential) GetToken(ctx context.Context, options policy.TokenRequestOptions) (*azcore.AccessToken, error) {
+	var account public.Account
+	if len(c.client.Accounts()) > 0 {
+		account = c.client.Accounts()[0]
+	}
+	result, err := c.client.AcquireTokenSilent(ctx, options.Scopes, public.WithSilentAccount(account))
+	if err != nil {
+		result, err = c.client.AcquireTokenInteractive(ctx, options.Scopes)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
These two calls will automatically use/update the token by using `token.Replace` and `tocken.Export`, right?